kokkos@3.5.00 +sycl build fails: Kokkos_SYCL_Team.hpp:112:17: error: no member named 'group_broadcast' in namespace 'sycl'

### Steps to reproduce the issue

`kokkos +sycl std=17 %dpcpp` build fails using:
* `spack@develop` (783615cacfdec7bd712cd903d409bcef6fc7cef4 from `Mon Feb 14 21:23:35 2022 +0100`)
* Ubuntu 20.04, x86_64
* Intel DPCPP 2022.0.0 from the latest Intel OneAPI Toolkit

Compiler entry for `dpcpp`
```
- compiler:
    spec: dpcpp@2022.0.0
    paths:
      cc: /opt/intel/oneapi/compiler/2022.0.1/linux/bin/icx
      cxx: /opt/intel/oneapi/compiler/2022.0.1/linux/bin/dpcpp
      f77: /opt/intel/oneapi/compiler/2022.0.1/linux/bin/ifx
      fc: /opt/intel/oneapi/compiler/2022.0.1/linux/bin/ifx
    flags: {}
    operating_system: ubuntu20.04
    target: x86_64
    modules:
    - compiler/2022.0.1
    environment: {}
    extra_rpaths: []
```

**Concrete spec:** [kokkos.spec.json.txt](https://github.com/spack/spack/files/8092402/kokkos.spec.json.txt)

**Concretization**
```
 -   camgkdo  kokkos@3.5.00%dpcpp@2022.0.0~aggressive_vectorization~compiler_warnings~cuda~cuda_constexpr~cuda_lambda~cuda_ldg_intrinsic~cuda_relocatable_device_code~cuda_uvm~debug~debug_bounds_check~debug_dualview_modify_check~deprecated_code~examples~explicit_instantiation~hpx~hpx_async_dispatch~hwloc~ipo~memkind~numactl~openmp~pic+profiling~profiling_load_print~pthread~qthread~rocm+serial+shared+sycl~tests~tuning~wrapper build_type=RelWithDebInfo std=17 arch=linux-ubuntu20.04-x86_64
[+]  4oedsu6      ^cmake@3.22.2%gcc@9.3.0~doc+ncurses+openssl+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
[+]  nxjjpha          ^ncurses@6.2%gcc@9.3.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
[+]  osgk6q3              ^pkgconf@1.8.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]  qej6s6n          ^openssl@1.1.1m%gcc@9.3.0~docs certs=system arch=linux-ubuntu20.04-x86_64
[+]  mdh4pnq              ^perl@5.34.0%gcc@9.3.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
[+]  hhpgpwj                  ^berkeley-db@18.1.40%gcc@9.3.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-ubuntu20.04-x86_64
[+]  acrm735                  ^bzip2@1.0.8%gcc@9.3.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
[+]  4rwlz7r                      ^diffutils@3.8%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]  zth4ohv                          ^libiconv@1.16%gcc@9.3.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  6zhwknl                  ^gdbm@1.19%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]  uo5pdkn                      ^readline@8.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]  ufqpqnp                  ^zlib@1.2.11%gcc@9.3.0+optimize+pic+shared arch=linux-ubuntu20.04-x86_64
```

**Install output**
```
$> . /opt/intel/oneapi/setvars.sh

$> spack install --dirty -f ./kokkos.spec.json
...
==> Installing kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd
==> No binary for kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd found: installing from source
==> Warning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check dpcpp@2022.0.0 for further details]
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/74/748f06aed63b1e77e3653cd2f896ef0d2c64cb2e2d896d9e5a57fec3ff0244ff.tar.gz
==> No patches needed for kokkos
==> kokkos: Executing phase: 'cmake'
==> kokkos: Executing phase: 'build'
...
     1792    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/Kokkos_Atomics_Desul_Wrapper.hpp:6:
     1793    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics.hpp:14:
     1794    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Atomic_Ref.hpp:17:
     1795    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Generic.hpp:16:
     1796    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Compare_Exchange.hpp:33:
     1797    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Compare_Exchange_SYCL.hpp:12:
  >> 1798    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/SYCLConversions.hpp:37:20: error: no member named 'ON
             EAPI' in namespace 'sycl'
     1799      static constexpr DESUL_SYCL_NAMESPACE::memory_order value = DESUL_SYCL_NAMESPACE::memory_order::relaxed;
     1800                       ^~~~~~~~~~~~~~~~~~~~
     1801    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Macros.hpp:38:36: note: expanded from macro 'DESUL_SY
             CL_NAMESPACE'
     1802    #define DESUL_SYCL_NAMESPACE sycl::ONEAPI
     1803                                 ~~~~~~^
     1804    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/impl/Kokkos_HostSpace.cpp:102:

     ...

     1806    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/Kokkos_Atomics_Desul_Wrapper.hpp:6:
     1807    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics.hpp:14:
     1808    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Atomic_Ref.hpp:17:
     1809    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Generic.hpp:16:
     1810    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Compare_Exchange.hpp:33:
     1811    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Compare_Exchange_SYCL.hpp:12:
  >> 1812    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/SYCLConversions.hpp:29:63: error: no member named 'ON
             EAPI' in namespace 'sycl'
     1813      static constexpr DESUL_SYCL_NAMESPACE::memory_order value = DESUL_SYCL_NAMESPACE::memory_order::release;
     1814                                                                  ^~~~~~~~~~~~~~~~~~~~
     1815    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Macros.hpp:38:36: note: expanded from macro 'DESUL_SY
             CL_NAMESPACE'
     1816    #define DESUL_SYCL_NAMESPACE sycl::ONEAPI
     1817                                 ~~~~~~^
     1818    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/impl/Kokkos_SharedAlloc.cpp:45:
...
  >> 2945    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Compare_Exchange_SYCL.hpp:22:3: error: no member name
             d 'ONEAPI' in namespace 'sycl'
     2946      DESUL_SYCL_NAMESPACE::atomic_fence(DesulToSYCLMemoryOrder<MemoryOrder>::value,
     2947      ^~~~~~~~~~~~~~~~~~~~
     2948    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Macros.hpp:38:36: note: expanded from macro 'DESUL_SY
             CL_NAMESPACE'
     2949    #define DESUL_SYCL_NAMESPACE sycl::ONEAPI
     2950                                 ~~~~~~^
     2951    20 errors generated.
  >> 2952    make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/build.make:233: core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Profiling.cpp.o] Error 1
     2953    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/impl/Kokkos_ExecPolicy.cpp:45:
     2954    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/Kokkos_Core.hpp:56:
     2955    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-build-camgkdo/KokkosCore_Config_DeclareBackend.hpp:47:
     2956    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/decl/Kokkos_Declare_SERIAL.hpp:49:
     2957    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/Kokkos_Serial.hpp:59:
     2958    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/Kokkos_Parallel.hpp:54:

     ...

     2962    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/Kokkos_Atomic.hpp:77:
     2963    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/Kokkos_Atomics_Desul_Wrapper.hpp:6:
     2964    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics.hpp:14:
     2965    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Atomic_Ref.hpp:17:
     2966    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Generic.hpp:16:
     2967    In file included from /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Compare_Exchange.hpp:33:
  >> 2968    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Compare_Exchange_SYCL.hpp:30:3: error: no member name
             d 'ONEAPI' in namespace 'sycl'
     2969      DESUL_SYCL_NAMESPACE::atomic_ref<
     2970      ^~~~~~~~~~~~~~~~~~~~
     2971    /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-src/core/src/desul/atomics/Macros.hpp:38:36: note: expanded from macro 'DESUL_SY
             CL_NAMESPACE'
     2972    #define DESUL_SYCL_NAMESPACE sycl::ONEAPI
     2973                                 ~~~~~~^
     2974    fatal error: too many errors emitted, stopping now [-ferror-limit=]
  >> 2975    make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/build.make:163: core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace_deepcopy.cpp.o] Error 1
     2976    20 errors generated.
     2977    20 errors generated.
  >> 2978    make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/build.make:247: core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Serial.cpp.o] Error 1
  >> 2979    make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/build.make:93: core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o] Error 1
     2980    20 errors generated.
  >> 2981    make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/build.make:275: core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_SharedAlloc.cpp.o] Error 1
     2982    20 errors generated.
  >> 2983    make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/build.make:121: core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o] Error 1
     2984    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-build-camgkdo'
  >> 2985    make[1]: *** [CMakeFiles/Makefile2:956: core/src/CMakeFiles/kokkoscore.dir/all] Error 2
     2986    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-build-camgkdo'
  >> 2987    make: *** [Makefile:149: all] Error 2

See build log for details:
  /tmp/root/spack-stage/spack-stage-kokkos-3.5.00-camgkdordkdph2gcf4pwyipbogzef4rd/spack-build-out.txt
```





### Error message

<details><summary>Error message</summary><pre>
See above
</pre></details>


### Information on your system

* **Spack:** 0.17.1-1177-783615cacf
* **Python:** 3.9.7
* **Platform:** linux-ubuntu20.04-broadwell
* **Concretizer:** clingo

### Additional information

[spack-configure-args.txt](https://github.com/spack/spack/files/8092396/spack-configure-args.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/8092397/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/8092398/spack-build-out.txt)

@crtrott
@janciesko
@masterleinad
@DavidPoliakoff 
@wspear 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/kokkos/package.py
+++ b/var/spack/repos/builtin/packages/kokkos/package.py
@@ -47,6 +47,9 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("+rocm", when="@:3.0")
     conflicts("+sycl", when="@:3.3")
 
+    # https://github.com/spack/spack/issues/29052
+    conflicts("@:3.5.00 +sycl", when="%dpcpp@2022.0.0")
+
     tpls_variants = {
         'hpx': [False, 'Whether to enable the HPX library'],
         'hwloc': [False, 'Whether to enable the HWLOC library'],

